### PR TITLE
Create data source for service resource 

### PR DESCRIPTION
**Pull request by TomTucka**
Create a data source for services

### CODE REVIEW

**Comment by TomTucka** [comment]
Hey @heimweh,

I've added some documentation, let me know if you want anything else adding!

File: pagerduty/data_source_pagerduty_service_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	username := fmt.Sprintf("tf-%s", acctest.RandString(5))
+	email := fmt.Sprintf("%s@foo.com", username)
+	service := fmt.Sprintf("tf-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourcePagerDutyEscalationPolicyConfig(username, email, service),
```
Comment (heimweh) [inline pagerduty/data_source_pagerduty_service_test.go]: think this should be `testAccDataSourcePagerDutyServiceConfig`

File: pagerduty/data_source_pagerduty_service_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourcePagerDutyEscalationPolicy("pagerduty_service.test", "data.pagerduty_service.by_name"),
```
Comment (heimweh) [inline pagerduty/data_source_pagerduty_service_test.go]: think this should be `testAccDataSourcePagerDutyService`

File: website/docs/d/service.html.markdown [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The service name to use to find a service in the PagerDuty API.
+
+## Attributes Reference
+* `id` - The ID of the found service.
+* `name` - The short name of the found service.
+* `created_at`- Creation timestamp of the service
```
Comment (heimweh) [inline website/docs/d/service.html.markdown]: `created_at` and `status` can probably be removed

File: pagerduty/data_source_pagerduty_service_test.go (hunk 1 above)
Comment (TomTucka) [reply pagerduty/data_source_pagerduty_service_test.go]: Done 😄

File: pagerduty/data_source_pagerduty_service_test.go (hunk 2 above)
Comment (TomTucka) [reply pagerduty/data_source_pagerduty_service_test.go]: Done 😄

File: website/docs/d/service.html.markdown (hunk 3 above)
Comment (TomTucka) [reply website/docs/d/service.html.markdown]: Done 😄

File: pagerduty/data_source_pagerduty_service_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func testAccDataSourcePagerDutyServiceConfig(username, email, service string) string {
+	return fmt.Sprintf(`
+resource "pagerduty_user" "test" {
+  name  = "%s"
+  email = "%s"
+}
+
+resource "pagerduty_service" "test" {
```
Comment (heimweh) [inline pagerduty/data_source_pagerduty_service_test.go]: Think this resource must be present in the config:

```hcl
resource "pagerduty_escalation_policy" "test" {
  name        = "%s"
  num_loops   = 2
  rule {
    escalation_delay_in_minutes = 10
    target {
      type = "user_reference"
      id   = "${pagerduty_user.test.id}"
    }
  }
}
```

**Comment by heimweh** [comment]
Nice work @TomTucka! 👌
Left one more comment, but the tests should hopefully run after this 🙂

File: pagerduty/data_source_pagerduty_service_test.go (hunk 4 above)
Comment (TomTucka) [reply pagerduty/data_source_pagerduty_service_test.go]: Made these changes @heimweh, I can't run these tests locally for some reason and presumed they were being run on Travis!

**Comment by TomTucka** [comment]
@heimweh finally got the test working locally, all seems fine to me now! Happy for you to run them and check though 😁

**Pull request merged**